internal/errors.go

```diff
@@ -37,3 +37,23 @@ func (e *FatalError) Error() string {
 func (e *FatalError) Unwrap() error {
 	return e.Err
 }
+
+// PartialWriteError indicate that only a subset of the metrics were written
+// successfully (i.e. accepted). The rejected metrics should be removed from
+// the buffer without being successfully written. Please note: the metrics
+// are specified as indices into the batch to be able to reference tracking
+// metrics correctly.
+type PartialWriteError struct {
+	Err                 error
+	MetricsAccept       []int
+	MetricsReject       []int
+	MetricsRejectErrors []error
```

Comment:
Would it make sense to combine these into a single array, or are there cases where a metric could be rejected without an error here?

Reply:
I don't think we should combine those as we do not _always_ need to access the errors for rejected metrics, e.g. when updating the transaction, we don't use the error, whereas in the plugins you might want to log the errors without being interested in the metric indices...